com.spotify/index.js

```diff
@@ -0,0 +1,662 @@
+    let songs = string.split(',');
+    return songs;
+  }
+
+  async songNamesToURIs(songs){
+    let uris = [];
+    for (let i = 0; i < songs.length; i++){
+      let searchResults = await this.search(songs[i].trim(),'track');
+      //TODO: Possibly add more error checking here.
+      if (!searchResults.hasOwnProperty('tracks') || searchResults.tracks.total === 0) continue;
+      uris.push(searchResults.tracks.items[0].uri);
+    }
+    return uris;
+  }
+
+// plays multiple songs, separated by the phrase "comma"
+  async do_play_multiple_songs({toPlay}){      // rename this later to "play songs"
+    let songs = this.splitMultiString(toPlay);
+    console.log('songs to play are ' + JSON.stringify(songs));
+    let uris = await this.songNamesToURIs(songs);
+    console.log('uris are ' + JSON.stringify(uris));
+    return this.player_play_helper(JSON.stringify({'uris':uris}));
+  }
+
+  do_play_seek_seconds({seconds}) {
+    console.log("seeking to milliseconds: " + seconds);
+    let milliseconds = Math.round(seconds);
+    let seek_url = SEEK_URL + '?position_ms=' + milliseconds.toString();
+    return this.http_put_default_options(seek_url, '');
+  }
+
+  do_save_current_track() {
+    return this.currently_playing_helper().then(response => {
+      if (response === '' || response.length === 0 ) {
+        return;
+      }
+      const parsed = JSON.parse(response);
+      if (parsed.is_playing === false) {
+        return;
+      }
+      let id = parsed.item.id;
+      let save_url = TRACKS_URL + '?ids=' + id;
+      return this.http_put_default_options(save_url, '');
+    });
+  }
+
+  /* Audio Features */
+  audio_features_get_by_id(id) {
+    return Tp.Helpers.Http.get(AUDIO_FEATURES_URL + id, {
+      accept: 'application/json',
+      useOAuth2: this
+    });
+  }
+
+  audio_features_helper(songName, transformation) {
+    console.log("Audio features helper received songname: " + songName);
+    if (typeof(songName) == 'undefined' || songName === "") {
+      return this.currently_playing_helper().then(response => {
+        let NONE_RESULT = [{human_output: "No song is currently playing."}];
+        if (response === '' || response.length === 0 ) {
+          return NONE_RESULT;
+        }
+        const parsed = JSON.parse(response);
+        if (parsed.is_playing === false) {
+          return NONE_RESULT;
+        }
+        let id = parsed.item.id;
+        return this.audio_features_get_by_id(id).then(response => transformation(JSON.parse(response)));
+      });
+    } else {
+      this.search(songName,'track').then(searchResults=>{
+        //TODO: Possibly add more error checking here.
+        if (!searchResults.hasOwnProperty('tracks'))
+          return [{human_output: "No song found for that query."}];;
+        let id = searchResults.tracks.items[0].id;
+        return this.audio_features_get_by_id(id).then(response => transformation(JSON.parse(response)));
+      });
+    }
+  }
+
+  get_get_tempo({song}) {
```

Comment:
Because the underlying API is only one, all the get_* functions should be a single ThingTalk function with multiple parameters. Individual parameters can be selected in ThingTalk using a projection.
Having only one function will improve the quality of the natural language, and reduce the number of sentences you need.